sql: TestActiveCancelSession failed under stress

SHA: https://github.com/cockroachdb/cockroach/commits/99064bf2797cce49e2d56d0324a3344cf2312aca

Parameters:
```
TAGS=
GOFLAGS=
```

To repro, try:

```
# Don't forget to check out a clean suitable branch and experiment with the
# stress invocation until the desired results present themselves. For example,
# using stress instead of stressrace and passing the '-p' stressflag which
# controls concurrency.
./scripts/gceworker.sh start && ./scripts/gceworker.sh mosh
cd ~/go/src/github.com/cockroachdb/cockroach && \
stdbuf -oL -eL \
make stressrace TESTS=TestActiveCancelSession PKG=github.com/cockroachdb/cockroach/pkg/sql TESTTIMEOUT=5m STRESSFLAGS='-maxtime 20m -timeout 10m' 2>&1 | tee /tmp/stress.log
```

Failed test: https://teamcity.cockroachdb.com/viewLog.html?buildId=1082568&tab=buildLog

```
I190106 06:30:14.471847 80853 storage/stores.go:243  [n2] read 0 node addresses from persistent storage
I190106 06:30:14.471874 80853 storage/stores.go:262  [n2] wrote 1 node addresses to persistent storage
I190106 06:30:14.473342 80853 server/node.go:672  [n2] bootstrapped store [n2,s2]
I190106 06:30:14.473436 80853 server/node.go:546  [n2] node=2: started with [<no-attributes>=<in-mem>] engine(s) and attributes []
I190106 06:30:14.473626 80853 server/status/recorder.go:610  [n2] available memory from cgroups (8.0 EiB) exceeds system memory 16 GiB, using system memory
I190106 06:30:14.473657 80853 server/server.go:1849  [n2] Could not start heap profiler worker due to: directory to store profiles could not be determined
I190106 06:30:14.473714 80853 server/server.go:1551  [n2] starting https server at 127.0.0.1:42469 (use: 127.0.0.1:42469)
I190106 06:30:14.473734 80853 server/server.go:1553  [n2] starting grpc/postgres server at 127.0.0.1:37751
I190106 06:30:14.473748 80853 server/server.go:1554  [n2] advertising CockroachDB node at 127.0.0.1:37751
I190106 06:30:14.477468 80853 server/server.go:1607  [n2] done ensuring all necessary migrations have run
I190106 06:30:14.477501 80853 server/server.go:1610  [n2] serving sql connections
I190106 06:30:14.483652 81410 rpc/nodedialer/nodedialer.go:90  [n1] unable to connect to n2: unable to look up descriptor for n2
I190106 06:30:14.496878 80856 storage/stores.go:262  [n1] wrote 1 node addresses to persistent storage
I190106 06:30:14.522123 81396 sql/event_log.go:131  [n2] Event: "node_join", target: 2, info: {Descriptor:{NodeID:2 Address:127.0.0.1:37751 Attrs: Locality: ServerVersion:2.1-3 BuildTag:v2.2.0-alpha.20181217-435-g99064bf StartedAt:1546756214471736115 LocalityAddress:[] XXX_NoUnkeyedLiteral:{} XXX_sizecache:0} ClusterID:f7b0d59a-6661-4986-9eea-814db60c31d0 StartedAt:1546756214471736115 LastUp:1546756214471736115}
I190106 06:30:14.522134 81105 server/server_update.go:66  [n2] no need to upgrade, cluster already at the newest version
I190106 06:30:15.904162 80914 gossip/gossip.go:1513  [n1] node has connected to cluster via gossip
I190106 06:30:15.904316 80914 storage/stores.go:262  [n1] wrote 1 node addresses to persistent storage
I190106 06:30:16.888641 81235 gossip/gossip.go:1513  [n2] node has connected to cluster via gossip
I190106 06:30:16.888778 81235 storage/stores.go:262  [n2] wrote 1 node addresses to persistent storage
I190106 06:30:24.300523 81023 server/status/runtime.go:464  [n1] runtime stats: 198 MiB RSS, 692 goroutines, 30 MiB/62 MiB/111 MiB GO alloc/idle/total, 18 MiB/54 MiB CGO alloc/total, 0.0 CGO/sec, 0.0/0.0 %(u/s)time, 0.0 %gc (651x), 37 MiB/37 MiB (r/w)net
W190106 06:30:24.311132 81025 server/node.go:886  [n1,summaries] health alerts detected: {Alerts:[{StoreID:1 Category:METRICS Description:ranges.underreplicated Value:19 XXX_NoUnkeyedLiteral:{} XXX_sizecache:0}] XXX_NoUnkeyedLiteral:{} XXX_sizecache:0}
I190106 06:30:24.475509 81361 server/status/runtime.go:464  [n2] runtime stats: 199 MiB RSS, 692 goroutines, 19 MiB/70 MiB/111 MiB GO alloc/idle/total, 19 MiB/55 MiB CGO alloc/total, 0.0 CGO/sec, 0.0/0.0 %(u/s)time, 0.0 %gc (652x), 38 MiB/38 MiB (r/w)net
E190106 06:30:24.484169 81792 sql/distsqlrun/flow_registry.go:233  [intExec=read orphaned table leases] flow id:e71fd4ae-26cb-4478-be07-691c66e1b5ca : 1 inbound streams timed out after 10s; propagated error throughout flow
I190106 06:30:24.773598 81813 util/stop/stopper.go:536  quiescing; tasks left:
1      [async] closedts-subscription
1      [async] closedts-rangefeed-subscriber
I190106 06:30:24.773821 81812 util/stop/stopper.go:536  quiescing; tasks left:
1      [async] closedts-subscription
1      [async] closedts-rangefeed-subscriber
W190106 06:30:24.774224 81235 gossip/gossip.go:1499  [n2] no incoming or outgoing connections
I190106 06:30:24.774997 81813 util/stop/stopper.go:536  quiescing; tasks left:
1      [async] closedts-rangefeed-subscriber
I190106 06:30:24.777010 81812 util/stop/stopper.go:536  quiescing; tasks left:
1      [async] closedts-rangefeed-subscriber
--- FAIL: TestActiveCancelSession (10.54s)
	run_control_test.go:321: expected 2 sessions but found 3

```

SHA: https://github.com/cockroachdb/cockroach/commits/5ebfeec052f9cee4e63757defe7c9120643293db

Parameters:
```
TAGS=
GOFLAGS=
```

To repro, try:

```
# Don't forget to check out a clean suitable branch and experiment with the
# stress invocation until the desired results present themselves. For example,
# using stressrace instead of stress and passing the '-p' stressflag which
# controls concurrency.
./scripts/gceworker.sh start && ./scripts/gceworker.sh mosh
cd ~/go/src/github.com/cockroachdb/cockroach && \
stdbuf -oL -eL \
make stress TESTS=TestActiveCancelSession PKG=github.com/cockroachdb/cockroach/pkg/sql TESTTIMEOUT=5m STRESSFLAGS='-stderr=false -maxtime 20m -timeout 10m'
```

Failed test: https://teamcity.cockroachdb.com/viewLog.html?buildId=1174912&tab=buildLog

```
I190313 05:39:36.916013 99938 server/node.go:546  [n2] node=2: started with [<no-attributes>=<in-mem>] engine(s) and attributes []
I190313 05:39:36.916290 99938 server/status/recorder.go:609  [n2] available memory from cgroups (8.0 EiB) exceeds system memory 16 GiB, using system memory
I190313 05:39:36.916327 99938 server/server.go:1828  [n2] Could not start heap profiler worker due to: directory to store profiles could not be determined
I190313 05:39:36.916403 99938 server/server.go:1533  [n2] starting https server at 127.0.0.1:39467 (use: 127.0.0.1:39467)
I190313 05:39:36.916423 99938 server/server.go:1535  [n2] starting grpc/postgres server at 127.0.0.1:35769
I190313 05:39:36.916440 99938 server/server.go:1536  [n2] advertising CockroachDB node at 127.0.0.1:35769
I190313 05:39:36.918803 99938 server/server.go:1589  [n2] done ensuring all necessary migrations have run
I190313 05:39:36.918832 99938 server/server.go:1592  [n2] serving sql connections
I190313 05:39:36.928499 100490 rpc/nodedialer/nodedialer.go:95  [n1] unable to connect to n2: unable to look up descriptor for n2
I190313 05:39:36.936406 100239 sql/event_log.go:126  [n2] Event: "node_join", target: 2, info: {Descriptor:{NodeID:2 Address:{NetworkField:tcp AddressField:127.0.0.1:35769} Attrs: Locality: ServerVersion:2.1 BuildTag:v2.1.6-6-g5ebfeec StartedAt:1552455576914144735 LocalityAddress:[]} ClusterID:a03b7965-c313-48f9-8975-bb1d1918b558 StartedAt:1552455576914144735 LastUp:1552455576914144735}
I190313 05:39:36.936965 99941 storage/stores.go:261  [n1] wrote 1 node addresses to persistent storage
I190313 05:39:36.956044 100237 server/server_update.go:67  [n2] no need to upgrade, cluster already at the newest version
I190313 05:39:37.682421 100589 storage/replica_consistency.go:127  [n1,consistencyChecker,s1,r1/1:/M{in-ax}] triggering stats recomputation to resolve delta of {ContainsEstimates:true LastUpdateNanos:1552455576928355444 IntentAge:0 GCBytesAge:0 LiveBytes:-44602 LiveCount:-932 KeyBytes:-43412 KeyCount:-932 ValBytes:-1190 ValCount:-932 IntentBytes:0 IntentCount:0 SysBytes:0 SysCount:0}
I190313 05:39:38.645729 99978 gossip/gossip.go:1489  [n1] node has connected to cluster via gossip
I190313 05:39:38.645910 99978 storage/stores.go:261  [n1] wrote 1 node addresses to persistent storage
I190313 05:39:38.939590 100305 gossip/gossip.go:1489  [n2] node has connected to cluster via gossip
I190313 05:39:38.939784 100305 storage/stores.go:261  [n2] wrote 1 node addresses to persistent storage
I190313 05:39:46.640303 100182 server/status/runtime.go:465  [n1] runtime stats: 183 MiB RSS, 658 goroutines, 24 MiB/62 MiB/106 MiB GO alloc/idle/total, 18 MiB/49 MiB CGO alloc/total, 0.0 CGO/sec, 0.0/0.0 %(u/s)time, 0.0 %gc (707x), 45 MiB/45 MiB (r/w)net
W190313 05:39:46.661854 100184 server/node.go:886  [n1,summaries] health alerts detected: {Alerts:[{StoreID:1 Category:METRICS Description:ranges.underreplicated Value:1}]}
I190313 05:39:46.918661 100415 server/status/runtime.go:465  [n2] runtime stats: 184 MiB RSS, 658 goroutines, 29 MiB/57 MiB/106 MiB GO alloc/idle/total, 18 MiB/49 MiB CGO alloc/total, 0.0 CGO/sec, 0.0/0.0 %(u/s)time, 0.0 %gc (707x), 42 MiB/42 MiB (r/w)net
E190313 05:39:46.933877 100791 sql/distsqlrun/flow_registry.go:230  [intExec=read orphaned table leases] flow id:1f91685e-443a-4809-864e-54babfbd39a3 : 1 inbound streams timed out after 10s; propagated error throughout flow
I190313 05:39:47.274010 100647 util/stop/stopper.go:537  quiescing; tasks left:
1      [async] closedts-subscription
1      [async] closedts-rangefeed-subscriber
I190313 05:39:47.274436 100646 util/stop/stopper.go:537  quiescing; tasks left:
1      [async] closedts-subscription
1      [async] closedts-rangefeed-subscriber
W190313 05:39:47.274541 100305 gossip/gossip.go:1475  [n2] no incoming or outgoing connections
I190313 05:39:47.278220 100647 util/stop/stopper.go:537  quiescing; tasks left:
1      [async] closedts-rangefeed-subscriber
I190313 05:39:47.278337 100646 util/stop/stopper.go:537  quiescing; tasks left:
1      [async] closedts-rangefeed-subscriber
--- FAIL: TestActiveCancelSession (10.72s)
	run_control_test.go:321: expected 2 sessions but found 3

```

SHA: https://github.com/cockroachdb/cockroach/commits/df200cbf3f407dbf349aa601ff9036b4dff88e83

Parameters:
```
TAGS=
GOFLAGS=
```

To repro, try:

```
# Don't forget to check out a clean suitable branch and experiment with the
# stress invocation until the desired results present themselves. For example,
# using stressrace instead of stress and passing the '-p' stressflag which
# controls concurrency.
./scripts/gceworker.sh start && ./scripts/gceworker.sh mosh
cd ~/go/src/github.com/cockroachdb/cockroach && \
stdbuf -oL -eL \
make stress TESTS=TestActiveCancelSession PKG=github.com/cockroachdb/cockroach/pkg/sql TESTTIMEOUT=5m STRESSFLAGS='-stderr=false -maxtime 20m -timeout 10m'
```

Failed test: https://teamcity.cockroachdb.com/viewLog.html?buildId=1252932&tab=buildLog

```
I190421 05:47:06.482541 99780 server/server.go:1820  [n2] Could not start heap profiler worker due to: directory to store profiles could not be determined
I190421 05:47:06.482602 99780 server/server.go:1534  [n2] starting https server at 127.0.0.1:38905 (use: 127.0.0.1:38905)
I190421 05:47:06.482618 99780 server/server.go:1536  [n2] starting grpc/postgres server at 127.0.0.1:35235
I190421 05:47:06.482630 99780 server/server.go:1537  [n2] advertising CockroachDB node at 127.0.0.1:35235
W190421 05:47:06.482648 99780 jobs/registry.go:316  [n2] unable to get node liveness: node not in the liveness table
I190421 05:47:06.487761 99780 server/server.go:1590  [n2] done ensuring all necessary migrations have run
I190421 05:47:06.487781 99780 server/server.go:1593  [n2] serving sql connections
I190421 05:47:06.496334 100135 gossip/server.go:232  [n1] received initial cluster-verification connection from {tcp 127.0.0.1:35235}
I190421 05:47:06.502319 100073 rpc/nodedialer/nodedialer.go:95  [n1] unable to connect to n2: failed to resolve n2: unable to look up descriptor for n2
I190421 05:47:06.506102 99783 storage/stores.go:261  [n1] wrote 1 node addresses to persistent storage
I190421 05:47:06.511533 100155 server/server_update.go:67  [n2] no need to upgrade, cluster already at the newest version
I190421 05:47:06.512244 100157 sql/event_log.go:126  [n2] Event: "node_join", target: 2, info: {Descriptor:{NodeID:2 Address:{NetworkField:tcp AddressField:127.0.0.1:35235} Attrs: Locality: ServerVersion:2.1 BuildTag:v2.1.6-28-gdf200cb StartedAt:1555825626480122303 LocalityAddress:[]} ClusterID:f164fd5b-6b2f-4bc4-b7e2-60752a67690b StartedAt:1555825626480122303 LastUp:1555825626480122303}
I190421 05:47:07.376547 100528 storage/replica_consistency.go:127  [n1,consistencyChecker,s1,r1/1:/M{in-ax}] triggering stats recomputation to resolve delta of {ContainsEstimates:true LastUpdateNanos:1555825626489713954 IntentAge:0 GCBytesAge:0 LiveBytes:-44602 LiveCount:-932 KeyBytes:-43412 KeyCount:-932 ValBytes:-1190 ValCount:-932 IntentBytes:0 IntentCount:0 SysBytes:0 SysCount:0}
I190421 05:47:08.646778 99820 gossip/gossip.go:1489  [n1] node has connected to cluster via gossip
I190421 05:47:08.646986 99820 storage/stores.go:261  [n1] wrote 1 node addresses to persistent storage
I190421 05:47:08.751373 100122 gossip/gossip.go:1489  [n2] node has connected to cluster via gossip
I190421 05:47:08.751524 100122 storage/stores.go:261  [n2] wrote 1 node addresses to persistent storage
I190421 05:47:16.353688 100020 server/status/runtime.go:465  [n1] runtime stats: 189 MiB RSS, 656 goroutines, 28 MiB/57 MiB/105 MiB GO alloc/idle/total, 18 MiB/56 MiB CGO alloc/total, 0.0 CGO/sec, 0.0/0.0 %(u/s)time, 0.0 %gc (686x), 9.5 MiB/9.5 MiB (r/w)net
W190421 05:47:16.361997 100022 server/node.go:886  [n1,summaries] health alerts detected: {Alerts:[{StoreID:1 Category:METRICS Description:ranges.underreplicated Value:1}]}
I190421 05:47:16.487312 100261 server/status/runtime.go:465  [n2] runtime stats: 191 MiB RSS, 656 goroutines, 18 MiB/64 MiB/105 MiB GO alloc/idle/total, 18 MiB/57 MiB CGO alloc/total, 0.0 CGO/sec, 0.0/0.0 %(u/s)time, 0.0 %gc (687x), 11 MiB/11 MiB (r/w)net
E190421 05:47:16.504391 100545 sql/distsqlrun/flow_registry.go:230  [intExec=read orphaned table leases] flow id:69af454d-7693-4211-bbcd-9a3db4bcd4a6 : 1 inbound streams timed out after 10s; propagated error throughout flow
I190421 05:47:16.786747 100602 util/stop/stopper.go:537  quiescing; tasks left:
1      [async] closedts-subscription
1      [async] closedts-rangefeed-subscriber
I190421 05:47:16.787006 100601 util/stop/stopper.go:537  quiescing; tasks left:
1      [async] closedts-subscription
1      [async] closedts-rangefeed-subscriber
W190421 05:47:16.787150 100122 gossip/gossip.go:1475  [n2] no incoming or outgoing connections
I190421 05:47:16.787970 100601 util/stop/stopper.go:537  quiescing; tasks left:
1      [async] closedts-rangefeed-subscriber
I190421 05:47:16.788090 100602 util/stop/stopper.go:537  quiescing; tasks left:
1      [async] closedts-subscription
--- FAIL: TestActiveCancelSession (10.53s)
	run_control_test.go:321: expected 2 sessions but found 3

```

--- a/pkg/sql/run_control_test.go
+++ b/pkg/sql/run_control_test.go
@@ -206,7 +206,7 @@ func testCancelSession(t *testing.T, hasActiveSession bool) {
 
 	// Wait for node 2 to know about both sessions.
 	if err := retry.ForDuration(10*time.Second, func() error {
-		rows, err := conn2.QueryContext(ctx, "SHOW CLUSTER SESSIONS")
+		rows, err := conn2.QueryContext(ctx, "SELECT * FROM [SHOW CLUSTER SESSIONS] WHERE application_name NOT LIKE '$%'")
 		if err != nil {
 			return err
 		}

